django/db/backends/oracle/utils.py

```diff
@@ -27,11 +27,14 @@ class InsertVar:
     def __init__(self, field):
         internal_type = getattr(field, 'target_field', field).get_internal_type()
         self.db_type = self.types.get(internal_type, str)
+        self._bound_param = None
 
     def bind_parameter(self, cursor):
-        param = cursor.cursor.var(self.db_type)
-        cursor._insert_id_var = param
-        return param
+        self._bound_param = cursor.cursor.var(self.db_type)
```

Comment:
I don't know @felixxm. Binding variable to a cursor seems like a good idea to bein with. It's very implicit behavior and relies on the cursor to be consistent during a requiest lifecycle and threadsafe and when does that tuple get reset? How do we avoid injection or other side effects?

It makes more sense to me, to have `fetch_returned_insert_columns` receive parameters. Even if not all DBs work that way. I don't mind the API change, I am happy to document this, if we are actually to push this to 3.1. If we consider this a "bugfix" of the feature already included in 3.0, we don't even need to document anything, since the method is new.

You know my favorite, but you choose.

Reply:
It's not a bug but a new feature so we will not backport this patch to 3.0.

About implementation, I would like to get a second opinion on this. @charettes Can you take a look?

Reply:
I don't have a strong opinion either way.

I kind of like the explicitness of passing the parameters of `fetch_returned_insert_columns` but since `_insert_id_var` has been around for so long I see no harm and reusing this pattern. If there was issue WRT to concurrency they would likely already been surfaced.